packages/@react-stately/color/src/useHexColorFieldState.ts

```diff
@@ -0,0 +1,150 @@
+  const initialInputIsValid = isInputValueValid(colorInputValue, maxValue, minValue);
+  let colorValue;
+  let initialInputValue;
+  if (initialInputIsValid) {
+    colorValue = typeof colorInputValue === 'string'? new Color(colorInputValue) : colorInputValue;
+    initialInputValue = colorValue.toString('hex');
+  }
+  const [inputValue, setInputValue] = useState(initialInputValue || '');
+  const [isValid, setIsValid] = useState(initialInputIsValid);
+
+  const increment = () => {
+    setColorInputValue((previousValue) => {
+      const color = typeof previousValue === 'string'? new Color(previousValue) : previousValue;
+      const colorString = color.toString('hex').substring(1);
+      const colorNumber = parseInt(colorString, 16);
+
+      const maxColor = typeof maxValue === 'string'? new Color(maxValue) : maxValue;
+      const maxColorString = maxColor.toString('hex').substring(1);
+      const maxColorNumber = parseInt(maxColorString, 16);
+
+      const newValue = `#${Math.min(colorNumber + step, maxColorNumber).toString(16).toUpperCase()}`;
+      const newColor = new Color(newValue);
+      updateValidation(newColor);
+      setInputValue(newValue);
+      return newColor;
+    });
+  };
+
+  const incrementToMax = useCallback(() => {
+    if (maxValue != null) {
+      const maxColor = typeof maxValue === 'string'? new Color(maxValue) : maxValue;
+      setColorInputValue(maxColor);
+      setInputValue(maxColor.toString('hex'));
+    }
+  }, [maxValue, setColorInputValue, setInputValue]);
+
+  const decrement = () => {
+    setColorInputValue((previousValue) => {
+      const color = typeof previousValue === 'string'? new Color(previousValue) : previousValue;
+      const colorString = color.toString('hex').substring(1);
+      const colorNumber = parseInt(colorString, 16);
+
+      const minColor = typeof minValue === 'string'? new Color(minValue) : minValue;
+      const minColorString = minColor.toString('hex').substring(1);
+      const minColorNumber = parseInt(minColorString, 16);
+
+      const newValue = `#${Math.max(colorNumber - step, minColorNumber).toString(16).padStart(6, '0').toUpperCase()}`;
+      const newColor = new Color(newValue);
+      updateValidation(newColor);
+      setInputValue(newValue);
+      return newColor;
+    });
+  };
+
+  const decrementToMin = useCallback(() => {
+    if (minValue != null) {
+      const minColor = typeof minValue === 'string'? new Color(minValue) : minValue;
+      setColorInputValue(minColor);
+      setInputValue(minColor.toString('hex'));
+    }
+  }, [minValue, setColorInputValue, setInputValue]);
+
+  const setColorValue = (color: Color) => {
+    updateValidation(color);
+    setColorInputValue(color);
+    setInputValue(color.toString('hex'));
+  };
+
+  const updateValidation = (value: Color) => {
+    setIsValid(isInputValueValid(value, maxValue, minValue));
+  };
+
+  const commitInputValue = () => {
```

Comment:
I'm not quite sure how to write a test for this. I want to show how `colorValue` reverts back to its last valid value when it receives a new value through `setColorValue`. It's just that `setColorValue` strictly accepts `Color` objects and will always be valid afaik.

Reply:
Resolved. Added test `should revert to last valid value`